diff --git a/sub_mod_2/src/test/java/NaimTest.java b/sub_mod_2/src/test/java/NaimTest.java
index f3891e9..c0b695f 100644
--- a/sub_mod_2/src/test/java/NaimTest.java
+++ b/sub_mod_2/src/test/java/NaimTest.java
@@ -6,7 +6,11 @@ class NaimTest {
 
     Naim m = new Naim();
     @Test
-    void goo() {
+    void gooTest() {
+        assertEquals(m.goo(),0, "hey brother");
+    }
+    @Test
+    void newGooTest() {
         assertEquals(m.goo(),0, "hey brother");
     }
 }
\ No newline at end of file
